src/Providers/FlightsProvider/FlightsProvider.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {FC, ReactNode, useEffect, useState } from "react";
+import axios from "axios";
+import { EmptyFlight, Flight } from "typings/flights.d";
+
+interface IFlightContext {
+  flights: Flight[],
+  currentFlight: Flight,
+  switchCurrentFlight: (id: number) => void,
+  newest: Flight
+}
+
+export const FlightsContext = React.createContext<IFlightContext>({
+  flights: [],
+  currentFlight: EmptyFlight,
+  switchCurrentFlight: () => {},
+  newest: EmptyFlight,
+});
+
+// export const FlightsContext = React.createContext<IFlightContext | null>(null);
+
+const FlightsProvider: FC<{children: ReactNode}> = ({ children }) => {
+  const [flights, setFlights] = useState<Flight[]>([]);
+  const [currentFlight, setCurrentFlight] = useState<Flight>(EmptyFlight);
+  const [newest, setNewest] = useState<Flight>(EmptyFlight);
+  const [finishedFlights, setFinishedFlights] = useState<Flight[]>([]);
+
+  useEffect(() => {
+    axios
```

Comment:
If you would like to learn smth new it may be beneficial to replace 'raw' axios request with some react library for handling request. Eg. react-query (https://www.npmjs.com/package/@tanstack/react-query) or rtk-quer (https://redux-toolkit.js.org/rtk-query/overview).

This kind of libraries have bunch of usufl features like caching, paralel requests, loaders, etc